hydromt/models/components/grid.py

```diff
@@ -386,85 +178,20 @@ def read_grid(self, fn: str = "grid/grid.nc", **kwargs) -> None:
         **kwargs : dict
             Additional keyword arguments to be passed to the `read_nc` method.
         """
-        self._assert_read_mode()
+        if not self.root.is_reading_mode():
+            raise IOError("Model opened in write-only mode")
         self._initialize_grid(skip_read=True)
 
         # Load grid data in r+ mode to allow overwritting netcdf files
```

Comment:
`r+` does not overwrite, but append. is this what we want?

Reply:
Not sure, I did not change this code. Maybe @DirkEilander knows?